src/txn/txn_recover.c

```diff
@@ -474,6 +474,14 @@ __recovery_set_checkpoint_snapshot(WT_SESSION_IMPL *session)
     conn = S2C(session);
     counter = 0;
 
+    /*
+     * WiredTiger versions 10.0.1 onward have a valid checkpoint snapshot on-disk. Ignore reading
+     * the on-disk checkpoint snapshot from older versions.
+     */
+    if (conn->recovery_major < 10 ||
+      (conn->recovery_major == 10 && conn->recovery_minor == 0 && conn->recovery_patch == 0))
+        return (0);
```

Comment:
This feels like an easy to understand and low risk change. Do we also want to consider updating the patch number to 2 so this version gets released as 10.0.2? I don't know if that is a change that gets put in a PR here or is taken care of by @lukech at release time.

Reply:
I'm not sure how useful to bump the wiredtiger version for changes in this PR. In the past, the version was bumped when there's a data format change, log version bump, etc that correlated with compatibility requirements. And we don't have a standalone release planned for the near future. 

We should definitely have a separate conversation on how to update and make use of wiredtiger version in the future.

Reply:
The version number is bumped to 10.0.2 as a precautionary measure.